book/src/design/circuit/gadgets/decomposition.md

```diff
@@ -0,0 +1,43 @@
+# Decomposition
+Given a field element $\alpha$, these gadgets decompose it into $W$ $K$-bit windows $$\alpha = k_0 + 2^{K} \cdot k_1 + 2^{2K} \cdot k_2 + \cdots + 2^{(W-1)K} \cdot k_{W-1}$$ where each $k_i$ a $K$-bit value.
+
+This is done using a running sum $z_i, i \in [0..W).$ We initialize the running sum $z_0 = \alpha,$ and compute subsequent terms $z_{i+1} = \frac{z_i - k_i}{2^{K}}.$ This gives us:
+
+$$
+\begin{aligned}
+z_0 &= \alpha \\
+    &= k_0 + 2^{K} \cdot k_1 + 2^{2K} \cdot k_2 +  2^{3K} \cdot k_3 + \cdots, \\
+z_1 &= (z_0 - k_0) / 2^K \\
+    &= k_1 + 2^{K} \cdot k_2 +  2^{2K} \cdot k_3 + \cdots, \\
+z_2 &= (z_1 - k_1) / 2^K \\
+    &= k_2 +  2^{K} \cdot k_3 + \cdots, \\
+    &\vdots
```

Comment:
```suggestion
    &\vdots
z_W &= (z_{W-1} - k_{W-1}) / 2^K \\
    &= 0 \text{ (because } z_{W-1} = k_{W-1} \text{)}
```

Reply:
Technically this is only true in "Strict mode". Remind me again what the motivation was for having that as a mode, rather than always enforced?